### DISCUSSION TITLE
ChatHistory

### DISCUSSION THREAD

**Discussion by cherryymerryy** [egramtel/tdsharp#229] [Q&A]
Hi! How to get full chat history?

**Comment by ForNeVeR** [comment]
If you formulate the question in terms of tdlib, that would make it much easier to answer :(

This project provides bindings for tdlib, but we don't know all of the API ourselves.

**Reply by cherryymerryy** [reply]
if you use GetChatHistory, it will return only one message, and I need it to return all chat messages

**Reply by ForNeVeR** [reply]
What parameters do you pass to it?

**Reply by cherryymerryy** [reply]
```c#
var messages = await _client.ExecuteAsync(new TdApi.GetChatHistory
                {
                    ChatId = chatId,
                    Offset = offset,
                    Limit = 100
                });

**Reply by ForNeVeR** [reply]
Unfortunately, I cannot answer this. We've had some user questions about `GetChatHistory`, but it seems to be working for some other reporters. Perhaps ask the question upstream.
